internal/ethereum/deploy_contract_prepare.go

```diff
@@ -95,7 +96,9 @@ func (c *ethConnector) prepareDeployData(ctx context.Context, req *ffcapi.Contra
 	ethParams := make([]interface{}, len(req.Params))
 	for i, p := range req.Params {
 		if p != nil {
-			err := json.Unmarshal([]byte(*p), &ethParams[i])
+			decodedParam := json.NewDecoder(bytes.NewReader([]byte(*p)))
+			decodedParam.UseNumber()
+			err := decodedParam.Decode(&ethParams[i])
```

Comment:
I think we need to fix the issue in https://github.com/hyperledger/firefly-common/blob/main/pkg/fftypes/jsonany.go#L51 if the issue is how data gets unmarshalled

Reply:
Agreed - and it probably needs a quick search of the code for other places where `json.Unmarshal()` is used as well

Reply:
Going to prove out this fix with an end-to-end contract deploy first to check that the fix is good (beyond the UTs)

Reply:
Sorry @matthew1001 - I'm not sure I understand why there's a concern with `json.Unmarshal()`

Reply:
Ok - I do think I now understand, but the right fix (which I think would probably include `json.Unmarshal`) would take time to investigate